File: article-admin-main/src/api/task.ts
import { request } from '@/api/request.ts'
import type { Task } from '@/features/tasks/components/task-manager.tsx'

export function getTasks() {
  return request<Task[]>({ url: '/tasks' })
}

export function addTask(task: Task) {
  return request({ url: '/tasks', method: 'post', data: task })
}

export function updateTask(task: Task) {
  return request({ url: '/tasks', method: 'put', data: task })
}

export function deleteTask(task_id: number) {
  return request({ url: `/tasks/${task_id}`, method: 'delete' })
}

export function runTask(task_id: number) {
  return request<Task[]>({ url: `/tasks/run/${task_id}`, method: 'get' })
}

File: article-master/app/api/v1/task.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.api.deps import get_current_user
from app.api.services import task_service
from app.core.database import get_db
from app.models import User
from app.models.task import Task
from app.schemas.response import success
from app.schemas.task import TaskForm

router = APIRouter()


@router.get('/')
def list_task(db: Session = Depends(get_db), user: User = Depends(get_current_user)):
    return task_service.list_task(db)


@router.post('/')
def add_task(task_form: TaskForm, db: Session = Depends(get_db), user: User = Depends(get_current_user)):
    return task_service.add_task(db, task_form)


@router.put('/')
def update_task(task_form: TaskForm, db: Session = Depends(get_db), user: User = Depends(get_current_user)):
    return task_service.update_task(db, task_form)


@router.delete('/{task_id}')
def delete_task(task_id, db: Session = Depends(get_db), user: User = Depends(get_current_user)):
    return task_service.delete_task(db, task_id)


@router.get('/run/{task_id}')
def run_task(task_id, db: Session = Depends(get_db), user: User = Depends(get_current_user)):
    return task_service.run_task(db, task_id)
